packages/mui-material/src/InputBase/InputBase.js

```diff
@@ -130,7 +130,7 @@ export const InputBaseComponent = styled('input', {
   slot: 'Input',
   overridesResolver: inputOverridesResolver,
 })(({ theme, ownerState }) => {
-  const light = theme.palette.mode === 'light';
```

Comment:
Seems like this is used for the `opacity` value in the placeholder. I would propose maybe something similar as in the `Input` component (adding new token in the default theme for the `experimental_extendTheme` util).